### PR TITLE
Makes `styled-components` a peerDependency 💅

### LINKED ISSUE #35: Make `styled-components` a peerDependency?
A consumer of this package could want to use their own instance (or version) of `styled-components`. It'd be nice if they could install and provide their own instance of `styled-components`.

### PR DESCRIPTION
**Pull request by fiber-god**
Hello,

This PR makes `styled-components` a peer-dep. This is my first Status Open Bounty for me which is also pretty exciting. 

Happy OSS!

-- Juwan 

Fixes #35

### CODE REVIEW

**Comment by sohkai** [comment]
Hey Juwan (@fiber-god)!

Awesome, thanks for doing this! We'll fund the bounty tomorrow; hope you don't mind waiting a bit :).

It also might take us a bit of time to merge this in, since we have to make sure everything works correctly with our dependent projects (e.g. [aragon/aragon](https://github.com/aragon/aragon). We've been hesitating on doing this ourselves because we're not sure if everything will just work out of the box (fingers crossed).

**Comment by fiber-god** [comment]
I've removed the re-export, and added styled-components as a dependency for the gallery. I can move gallery's react, react-dom, and styled-components dep to peerdependencies too if needed, but testing how it works on the root project should probably be done first 👍

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ dist/
 lib/
 node_modules/
 npm-debug.log
+.vscode/
```
Comment (sohkai) [inline .gitignore]: Do you mind reverting this change? Editor-specific config should usually go into your global .gitignore :).

Comment (fiber-god) [reply .gitignore]: Yep! I actually didn't know that global .gitignore's were a thing, so thanks for the insight!

**Comment by fiber-god** [comment]
Those extra commits were generated from running a rebase on my current PR. I ran:

```
❯ git remote add upstream https://github.com/aragon/aragon-ui.git
❯ git fetch upstream
❯ git rebase upstream/master
```

It shouldn't affect my changes (it only keeps it in sync with Aragon's master), but I just wanted to explain why these commits appeared.

File: src/components/SidePanel/SidePanel.js [hunk 2]
```diff
@@ -62,9 +62,12 @@ const StyledPanelCloseButton = styled.button`
     cursor: pointer;
     background: none;
     border: 0;
+    outline: 0;
     &::-moz-focus-inner {
       border: 0;
     }
+
+    ${unselectable()};
```
Comment (sohkai) [inline src/components/SidePanel/SidePanel.js]: Hmm, not sure why these changes are showing up here if you rebased them. Can you try pulling and rebasing again?

Comment (fiber-god) [reply src/components/SidePanel/SidePanel.js]: just rebased and those changes don't seem to be in the latest push

Comment (sohkai) [reply src/components/SidePanel/SidePanel.js]: Hmm, now it's showing the newest SidePanel changes. Really weird...

Comment (fiber-god) [reply src/components/SidePanel/SidePanel.js]: That's how it should be right? Rebase replays my work on top of the newest `master` changes

**Comment by sohkai** [comment]
644e1059f8c65149470ca3214dc1a74142f69722 seems like it should probably be reverted?

File: src/components/SidePanel/SidePanel.js (hunk 2 above)
Comment (sohkai) [reply src/components/SidePanel/SidePanel.js]: Well, you're right, but the diff of the SidePanel changes shouldn't be showing up on Github. Not sure what's going on; my local git diff doesn't show it as being different.

Anyhow, don't worry about it :).

**Comment by sohkai** [comment]
Going to merge since it seems to be working seamlessly with https://github.com/aragon/aragon/pull/91 :).

Thanks @fiber-god for taking the effort to do this!
